lib/elixir/lib/kernel.ex

```diff
@@ -2973,7 +2973,9 @@ defmodule Kernel do
 
   translates to:
 
-      when x >= 1 and x <= 3
+      when is_integer(x) and x >= 1 and x <= 3
+
+  Note that a value can only be `in` a range when it's an integer
```

Comment:
This looks witty but not very clear IMO (using the `in` as both a word as the operator), what about saying:

> Note that a value can only be considered inside a range by `in` when such value is an integer.

?